sycl/plugins/unified_runtime/ur/adapters/hip/enqueue.cpp

```diff
@@ -82,6 +82,61 @@ void simpleGuessLocalWorkSize(size_t *ThreadsPerBlock,
     --ThreadsPerBlock[0];
   }
 }
+
+ur_result_t setHipMemAdvise(const void *DevPtr, size_t Size,
+                            ur_usm_advice_flags_t URAdviceFlags,
+                            hipDevice_t Device) {
+  std::unordered_map<ur_usm_advice_flags_t, hipMemoryAdvise>
+      URToHIPMemAdviseDeviceFlagsMap = {
+          {UR_USM_ADVICE_FLAG_SET_READ_MOSTLY, hipMemAdviseSetReadMostly},
+          {UR_USM_ADVICE_FLAG_CLEAR_READ_MOSTLY, hipMemAdviseUnsetReadMostly},
+          {UR_USM_ADVICE_FLAG_SET_PREFERRED_LOCATION,
+           hipMemAdviseSetPreferredLocation},
+          {UR_USM_ADVICE_FLAG_CLEAR_PREFERRED_LOCATION,
+           hipMemAdviseUnsetPreferredLocation},
+          {UR_USM_ADVICE_FLAG_SET_ACCESSED_BY_DEVICE,
+           hipMemAdviseSetAccessedBy},
+          {UR_USM_ADVICE_FLAG_CLEAR_ACCESSED_BY_DEVICE,
+           hipMemAdviseUnsetAccessedBy},
+      };
+  for (auto &FlagPair : URToHIPMemAdviseDeviceFlagsMap) {
+    if (URAdviceFlags & FlagPair.first) {
+      UR_CHECK_ERROR(hipMemAdvise(DevPtr, Size, FlagPair.second, Device));
+    }
+  }
+
+  static std::unordered_map<ur_usm_advice_flags_t, hipMemoryAdvise>
+      URToHIPMemAdviseHostFlagsMap = {
+          {UR_USM_ADVICE_FLAG_SET_PREFERRED_LOCATION_HOST,
+           hipMemAdviseSetPreferredLocation},
+          {UR_USM_ADVICE_FLAG_CLEAR_PREFERRED_LOCATION_HOST,
+           hipMemAdviseUnsetPreferredLocation},
+          {UR_USM_ADVICE_FLAG_SET_ACCESSED_BY_HOST, hipMemAdviseSetAccessedBy},
+          {UR_USM_ADVICE_FLAG_CLEAR_ACCESSED_BY_HOST,
+           hipMemAdviseUnsetAccessedBy},
+      };
+
+  for (auto &FlagPair : URToHIPMemAdviseHostFlagsMap) {
+    if (URAdviceFlags & FlagPair.first) {
+      UR_CHECK_ERROR(
+          hipMemAdvise(DevPtr, Size, FlagPair.second, hipCpuDeviceId));
+    }
+  }
+
+  static constexpr std::array<ur_usm_advice_flags_t, 4> UnmappedMemAdviceFlags =
+      {UR_USM_ADVICE_FLAG_SET_NON_ATOMIC_MOSTLY,
```

Comment:
If these are flags, then you don't need the array or the loop
```cpp
if (URAdviceFlags & (UR_USM_ADVICE_FLAG_SET_NON_ATOMIC_MOSTLY | 
                              UR_USM_ADVICE_FLAG_CLEAR_NON_ATOMIC_MOSTLY |
                              UR_USM_ADVICE_FLAG_BIAS_CACHED | 
                              UR_USM_ADVICE_FLAG_BIAS_UNCACHED) {
  return UR_RESULT_ERROR_INVALID_ENUMERATION;
}

Reply:
Yes. Another one for silly me. Cheers!